src/persist/src/indexed/encoding.rs

```diff
@@ -330,8 +340,26 @@ impl BlobTraceMeta {
     /// Asserts Self's documented invariants, returning an error if any are
     /// violated.
     pub fn validate(&self) -> Result<(), Error> {
+        let upper = self.ts_upper();
+        let min = Antichain::from_elem(Timestamp::minimum());
+
+        if self.since != min && !PartialOrder::less_than(&self.since, &upper) {
+            return Err(format!(
+                "invalid trace since {:?} at or in advance of trace upper {:?}",
+                self.since, upper
+            )
+            .into());
+        }
+
         let mut prev: Option<&Description<u64>> = None;
         for (desc, _) in self.batches.iter() {
+            if !PartialOrder::less_equal(desc.since(), &self.since) {
+                return Err(format!(
+                    "invalid batch since: {:?} in advance of trace since {:?}",
```

Comment:
should this also be "at or in advance of"?

Reply:
`less_equal` bites again! I wrote this to say `if !(desc.since <= self.since)` which implies that `desc.since > self.since` (in one dimension). glad im not the only one getting tripped up by this :)